src/style-property-map-readonly.js

```diff
@@ -14,27 +14,34 @@
 
 (function(shared, scope, testing) {
 
-  function ComputedStylePropertyMap(element) {
-    this._element = element;
+  function StylePropertyMapReadOnly(styleObject) {
+    this._styleObject = styleObject;
   }
 
-  ComputedStylePropertyMap.prototype =
-    Object.create(shared.StylePropertyMap.prototype);
+  StylePropertyMapReadOnly.prototype = {
+    append: function(property, value) {},
+    delete: function(property) {},
+    get: function(property) {},
+    getAll: function(property) {},
+    has: function(property) {},
+    set: function(property, value) {},
+    getProperties: function() {}
+  };
 
-  ComputedStylePropertyMap.prototype.append = function(property, value) {
-    throw new TypeError('ComputedStylePropertyMap is immutable');
+  StylePropertyMapReadOnly.prototype.append = function(property, value) {
```

Comment:
The modification functions should just not exist on StylePropertyMapReadOnly - they can just be added to StylePropertyMap (and the error can be "Not implemented yet" now instead).